Add several features and add MinIO

BBSRKhelp opened the pull request:
Add a feature to create a pet, species, breed, photos. add MinIO to the project

BBSRKhelp committed:
Add several features and add MinIO

Add a feature to create a pet, species, breed, photos. add MinIO to the project

BBSRKhelp commented:
Сначала не понял как делать 7.1 и решил начать со следующей задачи (на то время создание фичи по созданию пета), в надежде что так пойму как делать 7.1. Так получилось что сделал фичу по созданию пета и добавил minio  провайдера. Потом порядок заданий изменился и пришлось сделать еще фичу для вида и породы.

BBSRKhelp commented:
Вопрос 1: Как лучше назвать валидатор, хэндлер и command по созданию пета и породы.
-Пример: PetCreateCommand или PetAddCommand...
Вопрос 2: UnitOfWork нужен во всех хэндлерах где происходит сохранение в бд? обязательно там все оборачивать в try..catch..?
Вопрос 3: Добавил класс заменяющий оболочки для некоторых VO ValueObjectList, но с ним вылетал exception, связанный с тем что ef core не может сконфигурировать свойства ReadOnlyList, хотя при этом обычные оболочки также используют ReadOnlyList и все норм. Возможно уже был такой вопрос у кого то, не знаю.
Вопрос 4: Можно ли использовать логгер в Domain Layer для агрегатов и сущностей или в контроллерах?

KirillSachkov commented on PetFamily.Backend/src/PetFamily.API/Contracts/Breed/BreedCreateRequest.cs:
```diff
@@ -0,0 +1,11 @@
+
+namespace PetFamily.API.Contracts.Breed;
+
+public record BreedCreateRequest(string Name)
```
Лучше назвать CreateBreedRequest, сначала глагол

KirillSachkov commented on PetFamily.Backend/src/PetFamily.API/Contracts/Pet/PetCreateRequest.cs:
```diff
@@ -0,0 +1,54 @@
+
+namespace PetFamily.API.Contracts.Pet;
+
+public record PetCreateRequest(
```
лучше сделать CreatePetRequest

KirillSachkov replied:
аналогично для всех реквестов и команд

KirillSachkov commented on PetFamily.Backend/src/PetFamily.API/Controllers/SpeciesController.cs:
```diff
@@ -0,0 +1,53 @@
+    [HttpPost]
+    public async Task<ActionResult<Guid>> CreateAsync(
+       [FromServices] SpeciesCreateHandler handler,
+       [FromServices] SpeciesCreateCommandValidator commandValidator,
```
Команду нужно валидировать в слое с Application

KirillSachkov replied:
для остальных кейсов аналогично

KirillSachkov commented on PetFamily.Backend/src/PetFamily.Application/Commands/Volunteer/AddPet/PetCreateHandler.cs:
```diff
@@ -0,0 +1,126 @@
+    {
+        var transaction = await _unitOfWork.BeginTransactionAsync(cancellationToken);
+
+        try
```
try catch нужно использовать, когда нужно сделать какие-то возвратные действи, rollback произойдет автоматически если при коммите будет ошибка

KirillSachkov commented:
> Вопрос 1: Как лучше назвать валидатор, хэндлер и command по созданию пета и породы. -Пример: PetCreateCommand или PetAddCommand... Вопрос 2: UnitOfWork нужен во всех хэндлерах где происходит сохранение в бд? обязательно там все оборачивать в try..catch..? Вопрос 3: Добавил класс заменяющий оболочки для некоторых VO ValueObjectList, но с ним вылетал exception, связанный с тем что ef core не может сконфигурировать свойства ReadOnlyList, хотя при этом обычные оболочки также используют ReadOnlyList и все норм. Возможно уже был такой вопрос у кого то, не знаю. Вопрос 4: Можно ли использовать логгер в Domain Layer для агрегатов и сущностей или в контроллерах?

1 и 2 вопрос в ревью написал, 3 вопрос напиши в технические вопросы в чате, там должны помочь, так сразу не скажу. 4 вопрос в домене логировать ничего не нужно, во всех остальных местах можно

BBSRKhelp committed:
Rename files. Add a validator to the handler.

Rename files commands, handlers, validators, requests. Add a validator to the handlers and remove the validator in the controllers.
